### DISCUSSION TITLE
Allowing optional arrays in requests

### DISCUSSION THREAD

**Discussion by Synchro** [spatie/laravel-data#866] [Q&A]
I have an endpoint that accepts single data items, along the lines of:

```json
{
    "title" : "Never gonna give you up",
    "artist" : "Rick Astley"
}
```
and this is neatly mapped to a `Song` DTO, and works very nicely with route injection. However, *the same endpoint* may also receive an array of these items:

```json
[
    {
        "title" : "Never gonna give you up",
        "artist" : "Rick Astley"
    },
    {
        "title" : "White wedding",
        "artist" : "Billy Idol"
    }
]
```

Do I have to give up on the nice route injection features to be able to handle this? This is a W3C-defined endpoint, so I can't change how it works.

**Comment by soltmar** [comment]
I've got similar issue, I cannot find a way to use array wrapped data using "[from a Request](https://spatie.be/docs/laravel-data/v4/as-a-data-transfer-object/request-to-data-object)" method.
I have a WebHook which can receive both, a single item or array of items but with the difference that a single item is still wrapped in array. Would be nice to have a solution to unwrap single item from array, and a second to collect from array of items directly from dependency injection
